packages/snap/src/utils/validation.ts

```diff
@@ -0,0 +1,89 @@
+import { ethers } from 'ethers';
+import type { ChainConfig } from 'src/keyring';
+import { assert, define, object, optional, StructError } from 'superstruct';
+
+import { throwError } from './util';
+
+const EthereumAddress = define(
+  'EthereumAddress',
+  (value) => typeof value === 'string' && ethers.isAddress(value),
+);
+
+const Url = define('Url', (value) => {
+  const urlPattern =
+    /^(https?:\/\/)?[\w\\.-]+(:\d{2,6})?(\/[\\/\w \\.-]*)?(\?[\\/\w .\-=]*)?$/u;
```

Comment:
I see in the regex that you're considering backslashes and spaces (`[\w\\.-]+` and `(\/[\\/\w \\.-]*)?`), these are quite unusual for urls. Are you validating a specific case?

Also, take into consideration that [in urls spaces are encoded as `%20`](https://developers.google.com/maps/url-encoding#:~:text=For%20example%2C%20spaces%20in%20a,should%20be%20encoded%20as%20%2C%20.) so I'm wondering if having it in the regex is necessary.

Reply:
updated urlPattern
```ts
  const urlPattern =
    /^(https?:\/\/)?[\w.-]+(:\d{2,6})?(\/[\w\/.-]*)?(\?[\w\/.&=\-%]*)?$/u;
    ```
    * Removed the match for backslashes (\\) and spaces (now encoded as %20 or +)
    * Added the percent character (%) to both the path and query string parts